storelogs/main.go

```diff
@@ -0,0 +1,117 @@
+// pmm-agent
+// Copyright 2019 Percona LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package storelogs help to store logs
+package storelogs
+
+import (
+	"bytes"
+	"container/ring"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// LogsStore implement ring save logs.
+type LogsStore struct {
+	log   *ring.Ring
+	Entry *logrus.Entry
+	count int
+	m     sync.RWMutex
+}
+
+// Init initializes basic parameters.
+func (l *LogsStore) Init(entry *logrus.Entry) {
```

Comment:
we use another way to create and initialize objects through functions `New` or `NewSomething` which returns initialized object.